src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx

```diff
@@ -28,12 +30,16 @@ function VideoPopoverMenuContextProvider({children}: ChildrenProps) {
     );
 
     const downloadAttachment = useCallback(() => {
-        if (currentlyPlayingURL === null) {
+        if (videoPopoverMenuPlayerRef.current === null) {
             return;
         }
-        const sourceURI = addEncryptedAuthTokenToURL(currentlyPlayingURL);
+        const {source} = videoPopoverMenuPlayerRef.current?.props ?? {};
+        if (typeof source === 'number' || !source) {
+            return;
+        }
+        const sourceURI = addEncryptedAuthTokenToURL(source.uri);
```

Comment:
`source.uri` already has encryptedAuthToken appended. We are adding it again here,And because of this double `encryptedAuthToken` in the URL causes this issue: https://github.com/Expensify/App/issues/43117 in Safari on mWeb.

Reply:
@jayeshmangwani Does this mean we have to fix this issue again? Or did your follow up solution make sure to not cause a regression?

Reply:
We have fixed the issue in a PR, comment was just to notify that we could have avoided this issue